packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts

```diff
@@ -157,6 +205,20 @@ class MedicalDataService {
     })
   }
 
+  private joinReservoirChanges(): void {
+    const sortedDescPumpSettings = this.medicalData.pumpSettings.sort((pumpSettings1: PumpSettings, pumpSettings2: PumpSettings) => {
+      const date1 = Number(new Date(pumpSettings1.normalTime))
+      const date2 = Number(new Date(pumpSettings2.normalTime))
+      return date2 - date1
```

Comment:
you could use directly the pumpSettings1.epoch fields (timestamp value)
```suggestion
      return pumpSettings2.epoch- pumpSettings1.epoch
```

Or use the sortDatum private method of this class (sort order is different but it does the same thing).

Reply:
I hadn't seen the `sortDatum()` method, it's better now that I reuse it